File: src/main/java/com/github/bgomar/bgconsolelogger/toolwindow/setup/PropertiesConsoleLoggerToolSetup.java
package com.github.bgomar.bgconsolelogger.toolwindow.setup;

import com.github.bgomar.bgconsolelogger.tools.ConsoleLoggerSettings;
import com.intellij.openapi.options.ConfigurableUi;
import com.intellij.openapi.ui.DialogPanel;
import org.jetbrains.annotations.NotNull;


import javax.swing.*;

public class PropertiesConsoleLoggerToolSetup  extends AbstractToolSetup implements ConfigurableUi<ConsoleLoggerSettings> {
    private static JTextField propertiesConsoleLoggerTextField1 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField2 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField3 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField4 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField5 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField6 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField7 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField8 = new JTextField();
    private static JTextField propertiesConsoleLoggerTextField9 = new JTextField();

    private static JButton propertiesConsoleLoggerSaveButton = new JButton();
    private static JButton propertiesConsoleLoggerLoad2Button = new JButton();
    private static JButton propertiesConsoleLoggerLoad1Button = new JButton();
    private static JButton propertiesConsoleLoggerCancelButton = new JButton();

    private static JButton propertiesConsoleLoggerDefaultButton1 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton2 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton3 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton4 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton5 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton6 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton7 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton8 = new JButton();
    private static JButton propertiesConsoleLoggerDefaultButton9 = new JButton();

    public static DialogPanel ui;
    public int preset = 0;

    public PropertiesConsoleLoggerToolSetup(
            JTextField propertiesConsoleLoggerTextField1,
            JTextField propertiesConsoleLoggerTextField2,
            JTextField propertiesConsoleLoggerTextField3,
            JTextField propertiesConsoleLoggerTextField4,
            JTextField propertiesConsoleLoggerTextField5,
            JTextField propertiesConsoleLoggerTextField6,
            JTextField propertiesConsoleLoggerTextField7,
            JTextField propertiesConsoleLoggerTextField8,
            JTextField propertiesConsoleLoggerTextField9,
            JButton propertiesConsoleLoggerSaveButton,
            JButton propertiesConsoleLoggerLoad2Button,
            JButton propertiesConsoleLoggerLoad1Button,
            JButton propertiesConsoleLoggerCancelButton,
            JButton propertiesConsoleLoggerDefaultButton1,
            JButton propertiesConsoleLoggerDefaultButton2,
            JButton propertiesConsoleLoggerDefaultButton3,
            JButton propertiesConsoleLoggerDefaultButton4,
            JButton propertiesConsoleLoggerDefaultButton5,
            JButton propertiesConsoleLoggerDefaultButton6,
            JButton propertiesConsoleLoggerDefaultButton7,
            JButton propertiesConsoleLoggerDefaultButton8,
            JButton propertiesConsoleLoggerDefaultButton9
    ) {
        super();
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField1 = propertiesConsoleLoggerTextField1;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField2 = propertiesConsoleLoggerTextField2;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField3 = propertiesConsoleLoggerTextField3;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField4 = propertiesConsoleLoggerTextField4;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField5 = propertiesConsoleLoggerTextField5;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField6 = propertiesConsoleLoggerTextField6;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField7 = propertiesConsoleLoggerTextField7;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField8 = propertiesConsoleLoggerTextField8;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField9 = propertiesConsoleLoggerTextField9;

        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerSaveButton = propertiesConsoleLoggerSaveButton;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerLoad2Button = propertiesConsoleLoggerLoad2Button;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerLoad1Button = propertiesConsoleLoggerLoad1Button;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerCancelButton = propertiesConsoleLoggerCancelButton;

        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton1 = propertiesConsoleLoggerDefaultButton1;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton2 = propertiesConsoleLoggerDefaultButton2;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton3 = propertiesConsoleLoggerDefaultButton3;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton4 = propertiesConsoleLoggerDefaultButton4;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton5 = propertiesConsoleLoggerDefaultButton5;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton6 = propertiesConsoleLoggerDefaultButton6;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton7 = propertiesConsoleLoggerDefaultButton7;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton8 = propertiesConsoleLoggerDefaultButton8;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton9 = propertiesConsoleLoggerDefaultButton9;
    }

    public PropertiesConsoleLoggerToolSetup() {
        setup();
    }

    public void setup() {
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField1.setText(ConsoleLoggerSettings.getPattern(0));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField2.setText(ConsoleLoggerSettings.getPattern(1));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField3.setText(ConsoleLoggerSettings.getPattern(2));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField4.setText(ConsoleLoggerSettings.getPattern(3));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField5.setText(ConsoleLoggerSettings.getPattern(4));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField6.setText(ConsoleLoggerSettings.getPattern(5));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField7.setText(ConsoleLoggerSettings.getPattern(6));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField8.setText(ConsoleLoggerSettings.getPattern(7));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField9.setText(ConsoleLoggerSettings.getPattern(8));

        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton1.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField1.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_1 : ConsoleLoggerSettings.DEFAULT_PATTERN_10));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton2.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField2.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_2 : ConsoleLoggerSettings.DEFAULT_PATTERN_11));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton3.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField3.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_3 : ConsoleLoggerSettings.DEFAULT_PATTERN_12));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton4.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField4.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_4 : ConsoleLoggerSettings.DEFAULT_PATTERN_13));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton5.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField5.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_5 : ConsoleLoggerSettings.DEFAULT_PATTERN_14));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton6.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField6.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_6 : ConsoleLoggerSettings.DEFAULT_PATTERN_15));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton7.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField7.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_7 : ConsoleLoggerSettings.DEFAULT_PATTERN_16));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton8.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField8.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_8 : ConsoleLoggerSettings.DEFAULT_PATTERN_17));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerDefaultButton9.addActionListener(e -> PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField9.setText((preset == 1) ? ConsoleLoggerSettings.DEFAULT_PATTERN_9 : ConsoleLoggerSettings.DEFAULT_PATTERN_18));

        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerSaveButton.addActionListener(e -> {
            saveActiveLoggers();
            if (preset == 1) {
                save1Loggers();
            } else if (preset == 2) {
                save2Loggers();
            }
        });

        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerCancelButton.addActionListener(e -> {
            if (preset == 1) {
                load1Loggers();
            } else if (preset == 2)  {
                load2Loggers();
            }
        });

        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerLoad1Button.addActionListener(e -> {
            if (preset == 1) {
                save1Loggers();
            } else if (preset == 2) {
                save2Loggers();
            }
            load1Loggers();
        });

        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerLoad2Button.addActionListener(e -> {
            if (preset == 1) {
                save1Loggers();
            } else if (preset == 2) {
                save2Loggers();
            }
            load2Loggers();

        });
    }

    private void saveActiveLoggers() {
        ConsoleLoggerSettings.setPattern(0, propertiesConsoleLoggerTextField1.getText());
        ConsoleLoggerSettings.setPattern(1, propertiesConsoleLoggerTextField2.getText());
        ConsoleLoggerSettings.setPattern(2, propertiesConsoleLoggerTextField3.getText());
        ConsoleLoggerSettings.setPattern(3, propertiesConsoleLoggerTextField4.getText());
        ConsoleLoggerSettings.setPattern(4, propertiesConsoleLoggerTextField5.getText());
        ConsoleLoggerSettings.setPattern(5, propertiesConsoleLoggerTextField6.getText());
        ConsoleLoggerSettings.setPattern(6, propertiesConsoleLoggerTextField7.getText());
        ConsoleLoggerSettings.setPattern(7, propertiesConsoleLoggerTextField8.getText());
        ConsoleLoggerSettings.setPattern(8, propertiesConsoleLoggerTextField9.getText());
    }

    private void save1Loggers() {
        ConsoleLoggerSettings.setPattern(9, propertiesConsoleLoggerTextField1.getText());
        ConsoleLoggerSettings.setPattern(10, propertiesConsoleLoggerTextField2.getText());
        ConsoleLoggerSettings.setPattern(11, propertiesConsoleLoggerTextField3.getText());
        ConsoleLoggerSettings.setPattern(12, propertiesConsoleLoggerTextField4.getText());
        ConsoleLoggerSettings.setPattern(13, propertiesConsoleLoggerTextField5.getText());
        ConsoleLoggerSettings.setPattern(14, propertiesConsoleLoggerTextField6.getText());
        ConsoleLoggerSettings.setPattern(15, propertiesConsoleLoggerTextField7.getText());
        ConsoleLoggerSettings.setPattern(16, propertiesConsoleLoggerTextField8.getText());
        ConsoleLoggerSettings.setPattern(17, propertiesConsoleLoggerTextField9.getText());
    }

    private void save2Loggers() {
        ConsoleLoggerSettings.setPattern(18, propertiesConsoleLoggerTextField1.getText());
        ConsoleLoggerSettings.setPattern(19, propertiesConsoleLoggerTextField2.getText());
        ConsoleLoggerSettings.setPattern(20, propertiesConsoleLoggerTextField3.getText());
        ConsoleLoggerSettings.setPattern(21, propertiesConsoleLoggerTextField4.getText());
        ConsoleLoggerSettings.setPattern(22, propertiesConsoleLoggerTextField5.getText());
        ConsoleLoggerSettings.setPattern(23, propertiesConsoleLoggerTextField6.getText());
        ConsoleLoggerSettings.setPattern(24, propertiesConsoleLoggerTextField7.getText());
        ConsoleLoggerSettings.setPattern(25, propertiesConsoleLoggerTextField8.getText());
        ConsoleLoggerSettings.setPattern(26, propertiesConsoleLoggerTextField9.getText());
    }

    private void load1Loggers() {
        preset = 1;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField1.setText(ConsoleLoggerSettings.getPattern(9));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField2.setText(ConsoleLoggerSettings.getPattern(10));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField3.setText(ConsoleLoggerSettings.getPattern(11));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField4.setText(ConsoleLoggerSettings.getPattern(12));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField5.setText(ConsoleLoggerSettings.getPattern(13));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField6.setText(ConsoleLoggerSettings.getPattern(14));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField7.setText(ConsoleLoggerSettings.getPattern(15));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField8.setText(ConsoleLoggerSettings.getPattern(16));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField9.setText(ConsoleLoggerSettings.getPattern(17));
    }

    private void load2Loggers() {
        preset = 2;
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField1.setText(ConsoleLoggerSettings.getPattern(18));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField2.setText(ConsoleLoggerSettings.getPattern(19));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField3.setText(ConsoleLoggerSettings.getPattern(20));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField4.setText(ConsoleLoggerSettings.getPattern(21));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField5.setText(ConsoleLoggerSettings.getPattern(22));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField6.setText(ConsoleLoggerSettings.getPattern(23));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField7.setText(ConsoleLoggerSettings.getPattern(24));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField8.setText(ConsoleLoggerSettings.getPattern(25));
        PropertiesConsoleLoggerToolSetup.propertiesConsoleLoggerTextField9.setText(ConsoleLoggerSettings.getPattern(26));
    }


    @Override
    public void reset(@NotNull ConsoleLoggerSettings settings) {
        ui.reset();
    }

    @Override
    public boolean isModified(@NotNull ConsoleLoggerSettings settings) {
        return ui.isModified();
    }

    @Override
    public void apply(@NotNull ConsoleLoggerSettings settings) {
        ui.apply();
    }

    @Override
    public @NotNull JComponent getComponent() {
        return ui;
    }
}

File: src/main/kotlin/com/github/bgomar/consolelogger/toolWindow/setup/PropertiesConsoleLoggerToolSetup.kt
package com.github.bgomar.consolelogger.toolWindow.setup

import com.intellij.openapi.options.ConfigurableUi
import com.intellij.openapi.ui.DialogPanel
import org.jetbrains.annotations.NotNull
import javax.swing.*

open class PropertiesConsoleLoggerToolSetup() : AbstractToolSetup(), ConfigurableUi<ConsoleLoggerSettings> {

    private var propertiesConsoleLoggerPreset = 1

    companion object {
        var propertiesConsoleLoggerTextField1 = JTextField()
        var propertiesConsoleLoggerTextField2 = JTextField()
        var propertiesConsoleLoggerTextField3 = JTextField()
        var propertiesConsoleLoggerTextField4 = JTextField()
        var propertiesConsoleLoggerTextField5 = JTextField()
        var propertiesConsoleLoggerTextField6 = JTextField()
        var propertiesConsoleLoggerTextField7 = JTextField()
        var propertiesConsoleLoggerTextField8 = JTextField()
        var propertiesConsoleLoggerTextField9 = JTextField()
        var propertiesConsoleLoggerSaveButton = JButton()
        var propertiesConsoleLoggerLoad2Button = JButton()
        var propertiesConsoleLoggerLoad1Button = JButton()
        var propertiesConsoleLoggerCancelButton = JButton()

        var propertiesConsoleLoggerDefaultButton1 = JButton()
        var propertiesConsoleLoggerDefaultButton2 = JButton()
        var propertiesConsoleLoggerDefaultButton3 = JButton()
        var propertiesConsoleLoggerDefaultButton4 = JButton()
        var propertiesConsoleLoggerDefaultButton5 = JButton()
        var propertiesConsoleLoggerDefaultButton6 = JButton()
        var propertiesConsoleLoggerDefaultButton7 = JButton()
        var propertiesConsoleLoggerDefaultButton8 = JButton()
        var propertiesConsoleLoggerDefaultButton9 = JButton()

        var ui = DialogPanel()
    }

    constructor(
        propertiesConsoleLoggerTextField1: JTextField,
        propertiesConsoleLoggerTextField2: JTextField,
        propertiesConsoleLoggerTextField3: JTextField,
        propertiesConsoleLoggerTextField4: JTextField,
        propertiesConsoleLoggerTextField5: JTextField,
        propertiesConsoleLoggerTextField6: JTextField,
        propertiesConsoleLoggerTextField7: JTextField,
        propertiesConsoleLoggerTextField8: JTextField,
        propertiesConsoleLoggerTextField9: JTextField,
        propertiesConsoleLoggerSaveButton: JButton,
        propertiesConsoleLoggerLoad2Button: JButton,
        propertiesConsoleLoggerLoad1Button: JButton,
        propertiesConsoleLoggerCancelButton: JButton,
        propertiesConsoleLoggerDefaultButton1: JButton,
        propertiesConsoleLoggerDefaultButton2: JButton,
        propertiesConsoleLoggerDefaultButton3: JButton,
        propertiesConsoleLoggerDefaultButton4: JButton,
        propertiesConsoleLoggerDefaultButton5: JButton,
        propertiesConsoleLoggerDefaultButton6: JButton,
        propertiesConsoleLoggerDefaultButton7: JButton,
        propertiesConsoleLoggerDefaultButton8: JButton,
        propertiesConsoleLoggerDefaultButton9: JButton
    ) : this() {
        Companion.propertiesConsoleLoggerTextField1 = propertiesConsoleLoggerTextField1
        Companion.propertiesConsoleLoggerTextField2 = propertiesConsoleLoggerTextField2
        Companion.propertiesConsoleLoggerTextField3 = propertiesConsoleLoggerTextField3
        Companion.propertiesConsoleLoggerTextField4 = propertiesConsoleLoggerTextField4
        Companion.propertiesConsoleLoggerTextField5 = propertiesConsoleLoggerTextField5
        Companion.propertiesConsoleLoggerTextField6 = propertiesConsoleLoggerTextField6
        Companion.propertiesConsoleLoggerTextField7 = propertiesConsoleLoggerTextField7
        Companion.propertiesConsoleLoggerTextField8 = propertiesConsoleLoggerTextField8
        Companion.propertiesConsoleLoggerTextField9 = propertiesConsoleLoggerTextField9

        Companion.propertiesConsoleLoggerSaveButton = propertiesConsoleLoggerSaveButton
        Companion.propertiesConsoleLoggerLoad2Button = propertiesConsoleLoggerLoad2Button
        Companion.propertiesConsoleLoggerLoad1Button = propertiesConsoleLoggerLoad1Button
        Companion.propertiesConsoleLoggerCancelButton = propertiesConsoleLoggerCancelButton

        Companion.propertiesConsoleLoggerDefaultButton1 = propertiesConsoleLoggerDefaultButton1
        Companion.propertiesConsoleLoggerDefaultButton2 = propertiesConsoleLoggerDefaultButton2
        Companion.propertiesConsoleLoggerDefaultButton3 = propertiesConsoleLoggerDefaultButton3
        Companion.propertiesConsoleLoggerDefaultButton4 = propertiesConsoleLoggerDefaultButton4
        Companion.propertiesConsoleLoggerDefaultButton5 = propertiesConsoleLoggerDefaultButton5
        Companion.propertiesConsoleLoggerDefaultButton6 = propertiesConsoleLoggerDefaultButton6
        Companion.propertiesConsoleLoggerDefaultButton7 = propertiesConsoleLoggerDefaultButton7
        Companion.propertiesConsoleLoggerDefaultButton8 = propertiesConsoleLoggerDefaultButton8
        Companion.propertiesConsoleLoggerDefaultButton9 = propertiesConsoleLoggerDefaultButton9
    }

    fun setup() {
        loadActiveLoggers()
        load1Loggers()

        propertiesConsoleLoggerDefaultButton1.addActionListener {
            propertiesConsoleLoggerTextField1.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_1
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_10
            }
        }
        propertiesConsoleLoggerDefaultButton2.addActionListener {
            propertiesConsoleLoggerTextField2.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_2
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_11
            }
        }
        propertiesConsoleLoggerDefaultButton3.addActionListener {
            propertiesConsoleLoggerTextField3.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_3
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_12
            }
        }
        propertiesConsoleLoggerDefaultButton4.addActionListener {
            propertiesConsoleLoggerTextField4.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_4
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_13
            }
        }
        propertiesConsoleLoggerDefaultButton5.addActionListener {
            propertiesConsoleLoggerTextField5.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_5
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_14
            }
        }
        propertiesConsoleLoggerDefaultButton6.addActionListener {
            propertiesConsoleLoggerTextField6.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_6
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_15
            }
        }
        propertiesConsoleLoggerDefaultButton7.addActionListener {
            propertiesConsoleLoggerTextField7.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_7
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_16
            }
        }
        propertiesConsoleLoggerDefaultButton8.addActionListener {
            propertiesConsoleLoggerTextField8.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_8
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_17
            }
        }
        propertiesConsoleLoggerDefaultButton9.addActionListener {
            propertiesConsoleLoggerTextField9.text = if (propertiesConsoleLoggerPreset == 1) {
                ConsoleLoggerSettings.DEFAULT_PATTERN_9
            } else {
                ConsoleLoggerSettings.DEFAULT_PATTERN_18
            }
        }

        propertiesConsoleLoggerCancelButton.addActionListener {
            when (propertiesConsoleLoggerPreset) {
                1 -> {
                    load1Loggers()
                    saveActiveLoggers()
                }
                2 -> {
                    load2Loggers()
                    saveActiveLoggers()
                }
            }
        }

        propertiesConsoleLoggerLoad1Button.addActionListener {
            load1Loggers()
            saveActiveLoggers()
        }

        propertiesConsoleLoggerLoad2Button.addActionListener {
            load2Loggers()
            saveActiveLoggers()
        }

        propertiesConsoleLoggerSaveButton.addActionListener {
            when (propertiesConsoleLoggerPreset) {
                1 -> {
                    saveActiveLoggers()
                    save1Loggers()
                }
                2 -> {
                    saveActiveLoggers()
                    save2Loggers()
                }
            }
        }
    }
    private fun loadActiveLoggers() {
        propertiesConsoleLoggerTextField1.text = ConsoleLoggerSettings.getPattern(0)
        propertiesConsoleLoggerTextField2.text = ConsoleLoggerSettings.getPattern(1)
        propertiesConsoleLoggerTextField3.text = ConsoleLoggerSettings.getPattern(2)
        propertiesConsoleLoggerTextField4.text = ConsoleLoggerSettings.getPattern(3)
        propertiesConsoleLoggerTextField5.text = ConsoleLoggerSettings.getPattern(4)
        propertiesConsoleLoggerTextField6.text = ConsoleLoggerSettings.getPattern(5)
        propertiesConsoleLoggerTextField7.text = ConsoleLoggerSettings.getPattern(6)
        propertiesConsoleLoggerTextField8.text = ConsoleLoggerSettings.getPattern(7)
        propertiesConsoleLoggerTextField9.text = ConsoleLoggerSettings.getPattern(8)
    }

    private fun saveActiveLoggers() {
        ConsoleLoggerSettings.setPattern(0, propertiesConsoleLoggerTextField1.text)
        ConsoleLoggerSettings.setPattern(1, propertiesConsoleLoggerTextField2.text)
        ConsoleLoggerSettings.setPattern(2, propertiesConsoleLoggerTextField3.text)
        ConsoleLoggerSettings.setPattern(3, propertiesConsoleLoggerTextField4.text)
        ConsoleLoggerSettings.setPattern(4, propertiesConsoleLoggerTextField5.text)
        ConsoleLoggerSettings.setPattern(5, propertiesConsoleLoggerTextField6.text)
        ConsoleLoggerSettings.setPattern(6, propertiesConsoleLoggerTextField7.text)
        ConsoleLoggerSettings.setPattern(7, propertiesConsoleLoggerTextField8.text)
        ConsoleLoggerSettings.setPattern(8, propertiesConsoleLoggerTextField9.text)
    }

    private fun save1Loggers() {
        ConsoleLoggerSettings.setPattern(9, propertiesConsoleLoggerTextField1.text)
        ConsoleLoggerSettings.setPattern(10, propertiesConsoleLoggerTextField2.text)
        ConsoleLoggerSettings.setPattern(11, propertiesConsoleLoggerTextField3.text)
        ConsoleLoggerSettings.setPattern(12, propertiesConsoleLoggerTextField4.text)
        ConsoleLoggerSettings.setPattern(13, propertiesConsoleLoggerTextField5.text)
        ConsoleLoggerSettings.setPattern(14, propertiesConsoleLoggerTextField6.text)
        ConsoleLoggerSettings.setPattern(15, propertiesConsoleLoggerTextField7.text)
        ConsoleLoggerSettings.setPattern(16, propertiesConsoleLoggerTextField8.text)
        ConsoleLoggerSettings.setPattern(17, propertiesConsoleLoggerTextField9.text)
    }

    private fun save2Loggers() {
        ConsoleLoggerSettings.setPattern(18, propertiesConsoleLoggerTextField1.text)
        ConsoleLoggerSettings.setPattern(19, propertiesConsoleLoggerTextField2.text)
        ConsoleLoggerSettings.setPattern(20, propertiesConsoleLoggerTextField3.text)
        ConsoleLoggerSettings.setPattern(21, propertiesConsoleLoggerTextField4.text)
        ConsoleLoggerSettings.setPattern(22, propertiesConsoleLoggerTextField5.text)
        ConsoleLoggerSettings.setPattern(23, propertiesConsoleLoggerTextField6.text)
        ConsoleLoggerSettings.setPattern(24, propertiesConsoleLoggerTextField7.text)
        ConsoleLoggerSettings.setPattern(25, propertiesConsoleLoggerTextField8.text)
        ConsoleLoggerSettings.setPattern(26, propertiesConsoleLoggerTextField9.text)
    }

    private fun load1Loggers() {
        propertiesConsoleLoggerPreset = 1
        propertiesConsoleLoggerTextField1.text = ConsoleLoggerSettings.getPattern(9)
        propertiesConsoleLoggerTextField2.text = ConsoleLoggerSettings.getPattern(10)
        propertiesConsoleLoggerTextField3.text = ConsoleLoggerSettings.getPattern(11)
        propertiesConsoleLoggerTextField4.text = ConsoleLoggerSettings.getPattern(12)
        propertiesConsoleLoggerTextField5.text = ConsoleLoggerSettings.getPattern(13)
        propertiesConsoleLoggerTextField6.text = ConsoleLoggerSettings.getPattern(14)
        propertiesConsoleLoggerTextField7.text = ConsoleLoggerSettings.getPattern(15)
        propertiesConsoleLoggerTextField8.text = ConsoleLoggerSettings.getPattern(16)
        propertiesConsoleLoggerTextField9.text = ConsoleLoggerSettings.getPattern(17)
    }

    private fun load2Loggers() {
        propertiesConsoleLoggerPreset = 2
        propertiesConsoleLoggerTextField1.text = ConsoleLoggerSettings.getPattern(18)
        propertiesConsoleLoggerTextField2.text = ConsoleLoggerSettings.getPattern(19)
        propertiesConsoleLoggerTextField3.text = ConsoleLoggerSettings.getPattern(20)
        propertiesConsoleLoggerTextField4.text = ConsoleLoggerSettings.getPattern(21)
        propertiesConsoleLoggerTextField5.text = ConsoleLoggerSettings.getPattern(22)
        propertiesConsoleLoggerTextField6.text = ConsoleLoggerSettings.getPattern(23)
        propertiesConsoleLoggerTextField7.text = ConsoleLoggerSettings.getPattern(24)
        propertiesConsoleLoggerTextField8.text = ConsoleLoggerSettings.getPattern(25)
        propertiesConsoleLoggerTextField9.text = ConsoleLoggerSettings.getPattern(26)
    }

    override fun reset(@NotNull settings: ConsoleLoggerSettings) {
        ui.reset()
    }

    override fun isModified(@NotNull settings: ConsoleLoggerSettings): Boolean {
        return ui.isModified()
    }

    override fun apply(@NotNull settings: ConsoleLoggerSettings) {
        ui.apply()
    }

    override fun getComponent(): JComponent {
        return ui
    }
}
